### PR TITLE
Draw an octagon around the lock icon

### PR DESCRIPTION
**Pull request by SolitaryCipher**
I found the lock icon hard to see on some backgrounds, so I added a blue hexagon. Screenshot attached.

screenshot

### CODE REVIEW

File: unlock_indicator.c [hunk 1]
```diff
@@ -21,14 +21,19 @@
 #include "unlock_indicator.h"
 #include "xinerama.h"
 
-#define LOCK_SCALE 3.5
+#define sq3 1.73205080756887729352 /* sqrt(3) */
+#define sq2 1.41421356237
+
+#define LOCK_SCALE 5.0
 #define LOCK_RADIUS (25 * LOCK_SCALE)
-#define LOCK_CENTER (32 * LOCK_SCALE)
```
Comment (SolitaryCipher) [inline unlock_indicator.c]: The drawing are is too small at 32.

File: unlock_indicator.c [hunk 2]
```diff
@@ -90,7 +95,7 @@ pam_state_t pam_state;
  *
  */
 static double scaling_factor(void) {
-    const int dpi = (double)screen->height_in_pixels * 25.4 /
```
Comment (SolitaryCipher) [inline unlock_indicator.c]: Why not a nice round number?

File: unlock_indicator.c [hunk 3]
```diff
@@ -149,6 +154,40 @@ xcb_pixmap_t draw_image(uint32_t *resolution) {
         cairo_set_line_cap(ctx, CAIRO_LINE_CAP_ROUND);
         cairo_set_line_join(ctx, CAIRO_LINE_JOIN_ROUND);
 
+        /* draw the background octagon */
+        cairo_set_source_rgb(ctx, 
+            color_bg[0], color_bg[1], color_bg[2]);
+        cairo_set_line_width(ctx, 1);
+        cairo_move_to(ctx, ( (1 + sq2) * BG_SCALE)+LOCK_CENTER, (  1        * BG_SCALE)+LOCK_CENTER);
```
Comment (SolitaryCipher) [inline unlock_indicator.c]: magic numbers can be found through trigonometry.

**Comment by djdt** [comment]
I like it however I'm not going to merge because this branch is basically for my personal lock screen.
I can see how the icon might be hard to see on some backgrounds so I'll add some options for the lock icon color.

**Pull request closed without merging**